### PR TITLE
TP bases

### PR DESCRIPTION
**Pull request by j-bd**
Exercices TP_bases

@Ngoguey42
@ClementDelgrange

Thanks so much for your help

### CODE REVIEW

File: ex1.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import collections
+
+def compter_lettre(word, letter):
```
Comment (Ngoguey42) [inline ex1.py]: Always use english in you code

File: ex1.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	i = word.count(letter)
+	print(f'dans le mot "{word}" il y a "{i}" fois la lettre "{letter}"')
```
Comment (Ngoguey42) [inline ex1.py]: In python people use two spaces for indentation (not 4 spaces, nor 1 tab)

File: ex2.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from random import randint
+
+a = randint(0, 10)
+nb_coup = 0
+print('Le nombre mystère est une valeur entre 0 et 10. Entrez votre valeur :')
+
+val = 101
+while val != a :
+	val = int(input())
```
Comment (Ngoguey42) [inline ex2.py]: What if input returns a string that can't be converted to int? A program should never crash in an unexpected way

File: ex3_v2.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+while True :
+
+	answer = ask_to_user('Veuillez choisir une valeur de la liste suivante', liste_pfc)
+
+	choix_pc = random.choice(liste_pfc)
+	comparison(choix_pc, answer)
+
+	answer = ask_to_user('Voulez-vous recommencer ? ', liste_fin)
+	if answer == 'n' :
+		break
```
Comment (Ngoguey42) [inline ex3_v2.py]: A text file should always end with a new line

File: ex4_v4.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+def compter_lignes(chemin_fichier) :
```
Comment (Ngoguey42) [inline ex4_v4.py]: The name of the parameter lead the reader to think that is should be string, but it is a file stream. You should either rename your parameter. or take a string as it suggests

File: ex5_v0.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	print(f"Longueur de la ligne {ligne} = {longueur}")
+
+
+def distance(point1, point2) :
+	dist = round(((point2[0]-point1[0])**2 + (point2[1]-point1[1])**2)**0.5, 3)
+	print(f"La distance entre {point1} et {point2} = {dist}")
+	return dist
+
+#ligne = ((0, 0), (2, 3), (3, 4), (4, 4))
+print("Merci de saisir votre liste de points sous la forme x1,y1 x2,y2 ... : ")
```
Comment (Ngoguey42) [inline ex5_v0.py]: You should enclose your code starting from line 18 in a main() function, and call it at the end of the file. You will then notice that it breaks part of your code

File: ex5_v0.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	longueur = 0
+	while nb_pts > 1 :
+		resultat = distance (ligne[nb_pts-1], ligne[nb_pts-2])
+		nb_pts = nb_pts - 1
+		longueur = longueur + resultat
+	print(f"Longueur de la ligne {ligne} = {longueur}")
+
+
+def distance(point1, point2) :
+	dist = round(((point2[0]-point1[0])**2 + (point2[1]-point1[1])**2)**0.5, 3)
```
Comment (Ngoguey42) [inline ex5_v0.py]: Why do you round ? it seems unnecessary

File: ex5_v0.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
+
+def longueur_ligne():
+	nb_pts = len(ligne)
+	longueur = 0
+	while nb_pts > 1 :
```
Comment (Ngoguey42) [inline ex5_v0.py]: Why do you iterate from the end of the line ? It works but it is unusual.
You should use a "for x, y in line:" here, to be more explicit

File: ex5_v0.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+valeurs = saisie_utilisateur.split(' ')
+coordonnees = []
+ligne_tmp = []
+for x in range(0,len(valeurs)) :
+	coordonnees.append(valeurs[x].split(','))
+	
+for y in range(0, len(coordonnees)) :
+	tmp = []
+	for z in range(0, 2) :
+		tmp.append(int(coordonnees[y][z]))
```
Comment (Ngoguey42) [inline ex5_v0.py]: Why do you convert to int instead of float ?

File: ex1.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+	
+def compter_lettres(word):
+	i  = {
+		x: word.count(x) 
+		for x in word
+
```
Comment (Ngoguey42) [inline ex1.py]: You should clean more your code before sending it to review. Normalize your code, remove unnecessary empty lines/spaces/comments. Add necessary empty lines/spaces/comments

File: ex5_v0.py [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+def distance(point1, point2) :
+	dist = round(((point2[0]-point1[0])**2 + (point2[1]-point1[1])**2)**0.5, 3)
+	print(f"La distance entre {point1} et {point2} = {dist}")
+	return dist
+
+#ligne = ((0, 0), (2, 3), (3, 4), (4, 4))
+print("Merci de saisir votre liste de points sous la forme x1,y1 x2,y2 ... : ")
+saisie_utilisateur = input()
+
+valeurs = saisie_utilisateur.split(' ')
```
Comment (Ngoguey42) [inline ex5_v0.py]: As a bonus you should try to do the full input parsing with one .split(' '), and ones regex, in a list comprehension

**Comment by ClementDelgrange** [comment]
Tips: to move files from a directory to another, you can use `git mv old_path new_path`.
By this way, your changes are still recorded by git.

**Pull request merged**